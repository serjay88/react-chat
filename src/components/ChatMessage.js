import React from 'react';
import classnames from "classnames";
import { withStyles } from 'material-ui/styles';
import Typography from 'material-ui/Typography';
import Avatar from './Avatar';
import Paper from 'material-ui/Paper';

import titleInitials from '../utils/title-initials';

const styles = theme => ({
  messageWrapper: {
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    padding: `${theme.spacing.unit}px ${theme.spacing.unit * 3}px`,
  },
  messageWrapperFromMe: {
    justifyContent: 'flex-end',
  },
  message: {
    maxWidth: '70%',
    minWidth: '10%',
    padding: theme.spacing.unit,
    marginLeft: theme.spacing.unit * 2
  },
  messageFromMe: {
    marginRight: theme.spacing.unit * 2,
    backgroundColor: '#e6dcff',
  },
});

const ChatMessage = ({ classes, sender, content }) => {
  const isMessageFromMe = sender === 'me';

  const userAvatar = (
    <Avatar colorFrom={sender}>
      {titleInitials(sender)}
    </Avatar>
  );

  return (
    <div className={classnames(
      classes.messageWrapper,
      isMessageFromMe && classes.messageWrapperFromMe
    )}>
    {!isMessageFromMe && userAvatar}
      <Paper className={classnames(
        classes.message,
        isMessageFromMe && classes.messageFromMe
      )}>
        <Typography variant="caption">
          {sender}
        </Typography>
        <Typography variant="body1">
          {content}
        </Typography>
      </Paper>
    {isMessageFromMe && userAvatar}
    </div>
  );
};

export default withStyles(styles)(ChatMessage);
