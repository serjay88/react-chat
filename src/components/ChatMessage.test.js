/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatMessage from './ChatMessage';

const mockData = {
  content: '12345',
  sender: {
    _id: '12345',
    firstName: 'Name',
    lastName: 'Surname',
    username: 'username',
  },
  activeUser: {
    _id: '12345',
  },
  createdAt: '2018-03-29T20:32:34.303Z',
  statusMessage: true,
};

describe('<ChatList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatMessage {...mockData} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly ', () => {
    const tree = renderer.create(<ChatMessage {...mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
