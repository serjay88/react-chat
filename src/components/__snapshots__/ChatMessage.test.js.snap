// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList /> renders correctly  1`] = `
<div
  className="ChatMessage-messageWrapper-31"
>
  <p
    className="MuiTypography-root-37 MuiTypography-body1-46 ChatMessage-statusMessage-35"
  >
    <span
      className="MuiTypography-root-37 MuiTypography-caption-47 ChatMessage-statusMessageUser-36"
      style={
        Object {
          "color": "#ffc107",
        }
      }
    >
      Name Surname
    </span>
    12345
    <span
      className="MuiTypography-root-37 MuiTypography-caption-47"
    >
      2 days ago
    </span>
  </p>
</div>
`;
